src/event-replay/reverse-file-stream.ts

```diff
@@ -34,9 +42,16 @@ export class ReverseFileStream extends stream.Readable {
       // Split into lines to fill the `lineBuffer`
       this.remainder = buffer.toString('utf8') + this.remainder;
       this.lineBuffer = this.remainder.split(/\r?\n/);
+
       // Ignore empty/trailing lines, `readable.push('')` is not recommended
       this.lineBuffer = this.lineBuffer.filter(line => line.length > 0);
+      const remainderHasPrefixEnding = this.remainder.startsWith('\n');
       this.remainder = this.lineBuffer.shift() ?? '';
+
+      // Preserve the line-ending char for the remainder if one was at the read boundary
+      if (remainderHasPrefixEnding) {
+        this.remainder = '\n' + this.remainder;
+      }
```

Comment:
@rafaelcr found and fixed a bug here where multiple lines were merged and emitted if a read chunk starts with a `\n` character